api/cas-server-core-api-services/src/main/java/org/apereo/cas/services/ServicesManager.java

```diff
@@ -120,4 +121,24 @@
     default int count() {
         return 0;
     }
+
+    /**
+     * Return a list of services for the passed domain.
+     *
+     * @param domain
+     * @return list of services
+     */
+    default Collection<RegisteredService> getServicesForDomain(String domain) {
+        return Collections.EMPTY_LIST;
```

Comment:
I'd prefer to be on the safe side and simply return a new List, Set, etc. immutable collections as such as not serializable and I am sure somewhere down the road that might bite us, given they are pretty difficult to catch up front.